src/SparkFun_Ublox_Arduino_Library.cpp

```diff
@@ -1639,32 +1640,11 @@ boolean SFE_UBLOX_GPS::factoryDefault(uint16_t maxWait)
   return (sendCommand(&packetCfg, maxWait) == SFE_UBLOX_STATUS_DATA_SENT); // We are only expecting an ACK
 }
 
-//Given a group, ID and size, return the value of this config spot
-//The 32-bit key is put together from group/ID/size. See other getVal to send key directly.
-//Configuration of modern u-blox modules is now done via getVal/setVal/delVal, ie protocol v27 and above found on ZED-F9P
-uint8_t SFE_UBLOX_GPS::getVal8(uint16_t group, uint16_t id, uint8_t size, uint8_t layer, uint16_t maxWait)
```

Comment:
Just a comment... I'm glad this function is gone. I never liked it. Using createKey for those (rare?) occasions when we can calculate the key formulaically is the right thing to do. Nice touch...

Reply:
To be clear [it's still pretty much there](https://github.com/sparkfun/SparkFun_Ublox_Arduino_Library/pull/134/files?file-filters%5B%5D=.h&file-filters%5B%5D=.ino#diff-a8080db1af5eceab0f8b758ee29dc3df87965950ff5f9d328ceeb3ef1be551b9R565), I just pulled out the createKey bits so we could have getVal8/16/32 without duplicate code. I agree with you, I will *never* call with a group, ID, size, I'll use the full 32-bit const instead. But I figured I would leave it in. createKey() is obviously helpful but I'm happy to remove the getVal8/16/32(group, id, size) functions if you'd like.

Reply:
Oh boy... I'm all in favor of efficient code. And there are times when it is perfectly safe and valid to calculate a key - the port protocols being a prime example. But here's the catch... I know I can go to the Interface Description and quickly and easily look up the key for whatever parameter I want to get/set. And then I can search for that key in the library and examples to see if there's already a function using it. But if we use the formulaic method, we lose that ability. However... It might be best to leave them in just in case a user out there who loves getVal8(group, id, size) and who will cry if we take it away. Shame to destroy backward-compatibility to save a few bytes. On balance, I suggest we leave them in for now.